src/ocean/meta/values/VisitValue.d

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************************
+
+    Generic utility allowing to recursively visit every field in an aggregate
+    and possibly those indirectly reachable from them.
+
+    Intended as a common implementation base for various more domain-specific
+    utilities.
+
+    Copyright:
+        Copyright (c) 2018 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.values.VisitValue;
+
+import ocean.meta.traits.Basic;
+import ocean.meta.types.Arrays;
+import ocean.meta.types.Typedef;
+
+/*******************************************************************************
+
+    Recursively iterates over `value` fields and calls `visitor.visit` for each
+    one of those. `visitor.visit` is expected to be a template with the
+    following signature:
+
+    `bool visit ( T ) ( T* value )`
+
+    The return value indicates whether the value should be recursed into or
+    not.
+
+    Example:
+
+    ---
+    struct SumIntegersVisitor
+    {
+        /// Visitor can have any internal state it may need
+        long sum;
+
+        /// Returns: 'true' if recursion shall continue for this type
+        bool visit ( T ) ( T* value )
+        {
+            static if (isIntegerType!(T))
+                sum += *value;
+
+            return true;
+        }
+    }
+
+    SumIntegersVisitor visitor;
+    visitValue(some_value, visitor);
+    ---
+
+    Params:
+        value = root value to visit
+        visitor = struct providing templated `visit` method accepting single
+            argument which will be a pointer to visited element
+
+*******************************************************************************/
+
+public void visitValue ( T, Visitor ) ( ref T value, ref Visitor visitor )
+{
+    VisitImpl!(Visitor) impl;
+    impl.visitor = &visitor;
+    impl.visitAll(&value);
+}
+
+/*******************************************************************************
+
+    Implementation of visiting logic. Some types are currently not supported
+    because it is not clear what would recursing mean for them.
+
+*******************************************************************************/
+
+private struct VisitImpl ( Visitor )
+{
+    // Refer to the persistent instance of visitor struct in case implementation
```

Comment:
Non-standard internal comment format. (Also for `visitAll`.)

Reply:
What do you mean? This is not a ddoc comment.

Reply:
Right. It should be.

Reply:
Why? We don't have strict requirement to document private implementation details in a same way, do we?

Reply:
I've never been aware of any explicit exception.